marconi-core/src/Marconi/Core/Experiment/Indexer/SQLiteIndexer.hs

```diff
@@ -298,13 +318,24 @@ instance
         traverse_ rollbackTable (indexer ^. rollbackPlan)
     pure $ indexer & dbLastSync .~ p
 
-instance (MonadIO m) => IsSync m event SQLiteIndexer where
+instance
+  (MonadIO m, SQL.FromRow (Point event), HasGenesis (Point event))
+  => IsSync m event SQLiteIndexer
+  where
   lastSyncPoint indexer =
     pure $ indexer ^. dbLastSync
 
--- TODO This often results in the runtime error:
--- SQLite3 returned ErrorBusy while attempting to perform close: unable to close due to unfinalized statements or unfinished backups
--- Should use 'withConnection' instead.
+  lastSyncPoints :: Word -> SQLiteIndexer event -> m [Point event]
+  lastSyncPoints n indexer = do
+    let (GetLastSyncPointsQuery query limitParam) = indexer ^. lastSyncPointsQuery
+    resultE <- runExceptT $ runLastSyncQuery (indexer ^. connection) query limitParam n
+    case resultE of
+      -- TODO Revisit this. Should probably throw error. However, can't just add constraint
+      -- 'MonadError IndexerError event' because then it conflicts with other functions which have
+      -- 'MonadError (QueryError query) event)'
```

Comment:
We should write a helper to transition between the 2 as we needed it in different places (`runExceptT` to get the `QueryError`, `either` on the result `throwError` on a `Left`).

Reply:
Need some help on this one. I'll schedule some time with you.